File: Php/core/Navigation/NavigationModel.php
namespace core\Navigation{

	class NavigationModel {

		public $heading;
		public $headingLink;
		public $headingIsLinked;
		public $nodes;
		public $cssClass;
		public $dataRequire;

		public function __construct($heading = null, $nodes = null, $cssClass = null){
			nodes = array();
			if($heading != null){
				heading = $heading;
			}
			if($nodes != null){
				nodes = $nodes;
			}
			if($cssClass != null){
				cssClass = $cssClass;
			}
		}

		protected function addNode(NavigationNode $node){
			nodes[] = $node;
		}

	}

}

File: Web/Models/NavigationModels/NavigationModel.cs
using System.Collections.Generic;
using Web.Extensions;

namespace Web.Models.NavigationModels
{
	public abstract class NavigationModel : IViewModel
    {
	    public string Heading { get; protected set; }
	    public IList<NavigationNode> Nodes { get; protected set; }
	    public string CssClass { get; protected set; }
        public bool IsEmpty { get; protected set; }

	    protected NavigationModel()
	    {
            Nodes = new List<NavigationNode>();
	    }

        public virtual View GetView()
        {
            return new View("~/Views/Navigation/Navigation.cshtml");
        }
    }
}